### PR TITLE
Workaround BigTableIO to write Iterable of KV (#1057)

### PR DESCRIPTION
**Pull request by nevillelyh**
Rework of #1057

### CODE REVIEW

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+import org.apache.beam.sdk.transforms.windowing.Repeatedly;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class BigtableBulkWriter
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: Please add `populateDisplayData` where applicable.

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      recordsWritten = 0;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) throws Exception {
+      checkForFailures(failures);
+      final Iterable<KV<ByteString, Iterable<Mutation>>> elements = c.element();
+      final Iterator<KV<ByteString, Iterable<Mutation>>> iterator = elements.iterator();
+
+      while (iterator.hasNext()) {
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: if you do foreach here you will save one line of code.

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java (hunk 1 above)
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: There are testing missing for this transform.

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      }
+    }
+  }
+
+  static class AssignToShard extends DoFn<KV<ByteString, Iterable<Mutation>>,
+      KV<Long, KV<ByteString, Iterable<Mutation>>>> {
+
+    private final int numOfShards;
+
+    AssignToShard(int numOfShards) {
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: `final`

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      }
+      throw exception;
+    }
+
+    /**
+     * An exception that puts information about the failed record being written in its message.
+     */
+    class BigtableWriteException extends IOException {
+
+      public BigtableWriteException(KV<ByteString, Iterable<Mutation>> record, Throwable cause) {
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: `final` to the parameters

File: scio-bigtable/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableServiceHelper.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableServiceHelper.java]: 2018

File: scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java [hunk 6]
```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (ravwojdyla) [inline scio-bigtable/src/main/java/com/spotify/scio/bigtable/BigtableBulkWriter.java]: 2018

**Comment by andrisnoko** [comment]
@ravwojdyla @nevillelyh

File: scio-bigtable/src/test/java/com/spotify/scio/bigtable/BigtableBulkWriterTest.java [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+      assertThat(actual, containsInAnyOrder(kvs));
+    }
+  }
+
+  /**
+   * Hepler to test createBulkShards.
+   */
+  private static class TestPTransform extends
+                                      PTransform<PCollection<KV<ByteString, Iterable<Mutation>>>,
```
Comment (regadas) [inline scio-bigtable/src/test/java/com/spotify/scio/bigtable/BigtableBulkWriterTest.java]: 🎨 the formatting looks weird

**Pull request merged**